### DISCUSSION TITLE
Volume keep detaching & attaching to a pod

### DISCUSSION THREAD

**Discussion by jwenjian** [longhorn/longhorn#6457] [Q&A]
Hi,

I am facing a strange issue, one of the volume is keep detaching & attaching to a pod, and it never ends.

version info:

longhorn 1.4.0
k8s: 1.24.10

PV access mode is RWO.

The pod is a single replica deployment, due to the strange issue, the deployment is keep terminate failed pod and create new one.

![image](https://github.com/longhorn/longhorn/assets/25657798/0d340d78-61b5-4019-a1d7-741c5a1d2e5a)

I can also see many error logs in engine manager, such as :

```
	Line 43053: 2023-08-04T09:58:48.781989510+08:00 time="2023-08-04T01:58:48Z" level=error msg="failed to init frontend: device pvc-4fcd3420-cac3-429c-9291-662a6cdfb080: failed to delete target iqn.2019-10.io.longhorn:pvc-4fcd3420-cac3-429c-9291-662a6cdfb080: failed to execute: tgtadm [--lld iscsi --op delete --mode logicalunit --tid 1 --lun 1], output , stderr tgtadm: this logical unit is still active\n: exit status 22"

	Line 43055: 2023-08-04T09:58:48.782589100+08:00 [longhorn-instance-manager] time="2023-08-04T01:58:48Z" level=warning msg="Process update: pvc-4fcd3420-cac3-429c-9291-662a6cdfb080-e-713aedee: state error: errorMsg: exit status 1"

```

also errors in replica manager:

```
Line    8: 2023-08-04T07:54:26.420589067+08:00 [pvc-4fcd3420-cac3-429c-9291-662a6cdfb080-r-df32dfb5] time="2023-08-03T23:54:26Z" level=warning msg="Received signal interrupt to shutdown"
	Line    9: 2023-08-04T07:54:26.421155294+08:00 [pvc-4fcd3420-cac3-429c-9291-662a6cdfb080-r-df32dfb5] time="2023-08-03T23:54:26Z" level=warning msg="Starting to execute registered shutdown func github.com/longhorn/longhorn-engine/app/cmd.startReplica.func4"
```

The volume is working for a long time (~4 months), but it stop working in this few days.

Any possible reasons?

Thanks

**Comment by c3y1huang** [comment]
Typically, [Longhorn unbinds and closes the connections before deleting the Logical Unit Number (LUN)](https://github.com/longhorn/longhorn-engine/blob/v1.4.0/vendor/github.com/longhorn/go-iscsi-helper/iscsidev/iscsi.go#L346-L367).

Can you run `tgtadm --lld iscsi --op show --mode conn --tid 1` from the engine instance manager and show the output?

> The volume is working for a long time (~4 months), but it stop working in this few days.

Is there anything you can recall that might have triggered this unexpected behavior?

**Reply by jwenjian** [reply]
command output:

```bash
instance-manager-r-efe70391f3c77890d9e4afad22959c64:/ # tgtadm --lld iscsi --op show --mode conn --tid 1
tgtadm: failed to send request hdr to tgt daemon, Transport endpoint is not connected

```

```bash
instance-manager-e-efe70391f3c77890d9e4afad22959c64:/ # tgtadm --lld iscsi --op show --mode conn --tid 1
Session: 1984
    Connection: 0
        Initiator: iqn.1994-05.com.redhat:62d0e9aed78a

```

**Reply by c3y1huang** [reply]
Looks normal. No idea for now. Can you provide the support bundle and your environment context?
Can you remember any operation that is done before triggering this issue?
